src/OSPSuite.Core/Domain/Services/SimModelBatch.cs

```diff
@@ -79,7 +80,10 @@ public SimulationRunResults RunSimulation()
          {
             updateSimulationValues();
             _simModelSimulation.RunSimulation();
-            return new SimulationRunResults(success: true, warnings: WarningsFrom(_simModelSimulation), results: getResults());
+            var hasResults = simulationHasResults(_simModelSimulation);
+
+            return new SimulationRunResults(success: hasResults, warnings: WarningsFrom(_simModelSimulation), results: getResults(),
+               error: hasResults ? null : Error.SimulationDidNotProduceResults);
```

Comment:
why don't you create a method (something around the lines of...)
```
createSimulationResults(string errorFromException = null) {
var hasResults = simulationHasResults(_simModelSimulation);
var warnings = WarningsFrom(_simModelSimulation);
var error = errorFromException ??  (hasResults ? null : Error.SimulationDidNotProduceResults)
if(error==null){
return SimulationRunResults(success: true , warnings, results: getResults())
}

//error
return SimulationRunResults(false, warnings, error)
}
```
and call that in both instances?